### DISCUSSION TITLE
emqx_dashboard_not_found

### DISCUSSION THREAD

**Discussion by neilyoung** [emqx/emqx#12582] [General]
I'm wondering what these messages come from and what they mean in the end. I have port 80 open for certbot, but there is no way to access the EMQX dashboard that way. 

Can anybody shed some light please?

TIA

`2024-02-24T03:36:47.351773+00:00 [warning] msg: unexpected_api_access, mfa: emqx_dashboard_not_found:init/2(25), request: #{pid => <0.121296.0>,port => 80,scheme => <<"http">>,version => 'HTTP/1.1',path => <<"/public/index.php">>,host => <<"3.212.253.171">>,peer => {{127,0,0,1},44888},bindings => #{},sock => {{127,0,0,1},18083},ref => 'http:dashboard',cert => undefined,headers => #{<<"accept">> => <<"*/*">>,<<"accept-language">> => <<"en-US,en;q=0.9">>,<<"connection">> => <<"upgrade">>,<<"host">> => <<"3.212.253.171">>,<<"upgrade-insecure-requests">> => <<"1">>,<<"user-agent">> => <<"Mozilla/5.0 (Linux; Linux x86_64; en-US) Gecko/20100101 Firefox/122.0">>,<<"x-forwarded-for">> => <<"45.95.147.236">>,<<"x-forwarded-proto">> => <<"https">>,<<"x-real-ip">> => <<"45.95.147.236">>},method => <<"GET">>,body_length => 0,path_info => undefined,qs => <<"s=/index/\\think\\app/invokefunction&function=call_user_func_array&vars[0]=md5&vars[1][]=Hello">>,streamid => 1,has_body => false,host_info => undefined}
`

PS: Messages found in /var/log/emqx/emqx.log.1

**Comment by ieQu1** [comment]
It looks like you were hit by a vulnerability scanner, or a botnet:

```
path => <<"/public/index.php">>
```
```
s => <<"s=/index/\\think\\app/invokefunction&function=call_user_func_array&vars[0]=md5&vars[1][]=Hello">>,
```
EMQX doesn't contain any PHP code, but vulnerability scanners (and actual botnets) usually like to check this URL. Moreover, it looks like vulnerability exploitation attempt (again, EMQX doesn't any use PHP frameworks).

Conclusion: if EMQX endpoints are exposed to the Internet, they will get probed like this, it's just a fact of life. To secure yourself, please make sure to follow the guide on the dashboard security: https://www.emqx.io/docs/en/latest/dashboard/introduction.html#first-login
As well as other security guides: https://www.emqx.io/docs/en/latest/deploy/cluster/security.html https://www.emqx.io/docs/en/latest/access-control/security-guide.html

**Comment by neilyoung** [comment]
Thanks for elaborating

> Conclusion: if EMQX endpoints are exposed to the Internet, they will get probed like this, it's just a fact of life. To secure yourself, please make sure to follow the guide on the dashboard security: 

Sure, but none of the EMQX ports is exposed to the internet. 80 is open for certbot and 443 for WSS, except 8883 is open for MQTTS. That's it. I'm really wondering why this entry lands in the emqx log.

**Reply by neilyoung** [reply]
Thanks for the hint with the cookie. Wasn't aware of this.

**Reply by ieQu1** [reply]
Judging by the `x-forwarded-for` header, this request was forwarded to EMQX by the front proxy. Worth checking its config?

**Reply by neilyoung** [reply]
Yes, NGINX is in front. Here the slightly obfuscated config (did hide real domain). For me it looks ok. I'm forwarding anything for / to the EMQX dashboard and /mqtt is intended to be just used for WS.


/etc/nginx/sites-enabled/default 

```
server {
    listen 443 http2 ssl;
    listen [::]:443 http2 ssl;
    server_name mqtt.domain.com;

    ssl_session_cache shared:SSL:10m;
    ssl_session_timeout 10m;
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers HIGH:!aNULL:!MD5;
    ssl_certificate /etc/letsencrypt/live/mqtt.domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/mqtt.domain.com/privkey.pem; # managed by Certbot


    location / {
        proxy_pass              http://127.0.0.1:18083;
        proxy_read_timeout      90;
        proxy_set_header        Host $host;
        proxy_set_header        X-Real-IP $remote_addr;
        proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header        X-Forwarded-Proto $scheme;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }

    location /mqtt {
        proxy_pass              http://127.0.0.1:8083;
        proxy_read_timeout      90;
        proxy_set_header        Host $host;
        proxy_set_header        X-Real-IP $remote_addr;
        proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header        X-Forwarded-Proto $scheme;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }
}

```

EDIT: Removed the stupid SSL comment :)

**Reply by neilyoung** [reply]
So that explains how the entries came into the log. Could have found that by myself. Thanks for the pointer.

**Reply by neilyoung** [reply]
Didn't explicitly look for that for now, but does the dashboard allow for client certificate authentication?

**Reply by ieQu1** [reply]
Honestly, I'm not sure myself, I'm more of a CLI kind of person. Maybe someone who works with the dashboard more can chip in. This page can help, though: https://emqx.io/docs/en/latest/dashboard/authn.html

**Reply by neilyoung** [reply]
Yes, thanks. I'm aware of this and having applied two of these methods. But IMHO this is only used for the MQTT client authentication, not for the dashboard.

However, it's Sunday. Thanks for your help so far and enjoy the rest of the day :)

**Reply by id** [reply]
@neilyoung Dashboard does not support mutual TLS, however, since you are terminating TLS in nginx, you could setup mutual TLS there.

**Reply by neilyoung** [reply]
Thanks. Yes, I did that already.
